addons/binding/org.openhab.binding.neato/src/main/java/org/openhab/binding/neato/handler/NeatoHandler.java

```diff
@@ -0,0 +1,257 @@
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.OnOffType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Channel;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.neato.internal.NeatoRobot;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link NeatoHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Patrik Wimnell - Initial contribution
+ */
+public class NeatoHandler extends BaseThingHandler {
+
+    private Logger logger = LoggerFactory.getLogger(NeatoHandler.class);
+
+    private String vacuumSerialNumber;
+    private String vacuumSecret;
+    private String vacuumName;
+
+    private NeatoRobot mrRobot;
+
+    private int refreshTime;
+    private ScheduledFuture<?> refreshTask;
+    private static int DEFAULTREFRESHTIME = 60;
+
+    private ThingStatus thisStatus;
+
+    public NeatoHandler(Thing thing) {
+        super(thing);
+        thisStatus = ThingStatus.OFFLINE;
+    }
+
+    public ScheduledFuture<?> getRefreshTask() {
+        return this.refreshTask;
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (channelUID.getId().equals(org.openhab.binding.neato.NeatoBindingConstants.COMMAND)) {
+            logger.debug("Ok - will handle command for CHANNEL_COMMAND");
+
+            try {
+                mrRobot.sendCommand(command);
+            } catch (Exception exc) {
+                logger.error("Error while processing command from openHAB. Error: {}", exc.getMessage());
+            }
+
+        }
+        this.refreshStateAndUpdate();
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.info("Will boot up Neato Vacuum Cleaner binding!");
+
+        updateStatus(ThingStatus.OFFLINE);
+        thisStatus = ThingStatus.OFFLINE;
+
+        Configuration config = getThing().getConfiguration();
+        vacuumSecret = (String) config.get(CONFIG_SECRET);
+        vacuumName = (String) config.get(CONFIG_NAME);
+        vacuumSerialNumber = (String) config.get(CONFIG_SERIAL);
+
+        logger.info("Got config settings for {} with serial number: {}", vacuumName, vacuumSecret);
+
+        super.initialize();
+
+        try {
+            refreshTime = ((BigDecimal) config.get(CONFIG_REFRESHTIME)).intValue();
+            if (refreshTime <= 0) {
+                throw new IllegalArgumentException("Refresh time must be positive number!");
+            }
+        } catch (Exception e) {
```

Comment:
What exceptions do you expect here, I think that there are none which you can not avoid.